docs: guide for creating new releases (#1304)
penrose/penrose: CONTRIBUTING.md

If you hit any snags in the process, run into bugs, or just have questions,
please file an issue!

## Release

> Our repo uses [semantic versioning]() and maintains the same version number for all packages. Generally speaking, we release new versions whenever new features are introduced (PRs with `feat` tag). Here are the steps for creating new releases.

Reviewer: ```suggestion
Our repo uses [semantic versioning][] and maintains the same version number for all packages. Generally speaking, we release new versions whenever new features are introduced (PRs with `feat` tag). Here are the steps for creating new releases.
```

> - Make sure all PRs for the uncoming PR are merged. Switch to `main` and check `git status` to make sure it's clean and up-to-date.

Reviewer: ```suggestion
- Make sure all PRs for the upcoming release are merged. Switch to `main` and check `git status` to make sure it's clean and up-to-date.
```

- At repo root, run `yarn new-version` to create a new version.
- Run `yarn format` to clean up auto-generated file changes.

> - Create a new branch (`git checkout -b <version-number>`) from main and commit the changes.

Reviewer: ```suggestion
- Create a new branch (`git switch --create release-X.Y.Z`) from main and commit the changes.
```

> - Open a new PR with a title `chore: bump version to <version-number>` and merge after CI passes.

Reviewer: ```suggestion
- Open a new PR with a title `chore: bump version to X.Y.Z` and merge after CI passes.
```

> - Create a new [GitHub release]().

Reviewer: ```suggestion
- Create a new [GitHub release][].
```

If you hit any snags in the process, run into bugs, or just have questions,
please file an issue!

## Release

Our repo uses [semantic versioning][] and maintains the same version number for all packages. Generally speaking, we release new versions whenever new features are introduced (PRs with `feat` tag). Here are the steps for creating new releases.

- Make sure all PRs for the upcoming release are merged. Switch to `main` and check `git status` to make sure it's clean and up-to-date.
- At repo root, run `yarn new-version` to create a new version.

>   - Note that the this script does not modify the Rust crate version in `packages/optimizer/Cargo.{toml,lock}`. If the upcoming release involves `@penrose/optimizer`, they may require manual updates.

Reviewer: ```suggestion
  - Note that the this script does not modify the Rust crate version in `packages/optimizer/Cargo.{toml,lock}`, which currently must be manually updated.
```
